### ISSUE TITLE
Installation issue: py-torch@2.1.2

### ISSUE THREAD

**Issue by fabiosanger** [LLNL/spack#42237]
### Steps to reproduce the issue

```console
$ spack spec -I <spec>
spack spec -I py-torch
Input spec
--------------------------------
 -   py-torch

Concretized
--------------------------------
 -   py-torch@2.1.2%gcc@13.1.0~caffe2~cuda~debug+distributed+fbgemm+gloo+kineto~metal+mkldnn+mpi+nnpack+numa+numpy+onnx_ml+openmp+qnnpack~rocm+tensorpipe~test+valgrind+xnnpack build_system=python_pip arch=linux-ubuntu22.04-x86_64_v3
[^]      ^cmake@3.27.7%gcc@13.1.0~doc+ncurses+ownlibs build_system=generic build_type=Release arch=linux-ubuntu22.04-x86_64_v3
[^]          ^curl@8.4.0%gcc@13.1.0~gssapi~ldap+libidn2~librtmp~libssh~libssh2+nghttp2 build_system=autotools libs=shared,static tls=openssl arch=linux-ubuntu22.04-x86_64_v3
[^]              ^libidn2@2.3.4%gcc@13.1.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[^]                  ^libunistring@1.1%gcc@13.1.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[^]              ^nghttp2@1.57.0%gcc@13.1.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[^]              ^pkgconf@1.9.5%gcc@13.1.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[^]          ^ncurses@6.4%gcc@13.1.0~symlinks+termlib abi=none build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[^]          ^zlib-ng@2.1.4%gcc@13.1.0+compat+opt build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]      ^eigen@3.4.0%gcc@13.1.0~ipo build_system=cmake build_type=RelWithDebInfo generator=make arch=linux-ubuntu22.04-x86_64_v3
[^]          ^gmake@4.4.1%gcc@13.1.0~guile build_system=generic arch=linux-ubuntu22.04-x86_64_v3
[+]      ^fp16@2020-05-14%gcc@13.1.0~ipo build_system=cmake build_type=Release generator=ninja arch=linux-ubuntu22.04-x86_64_v3
[+]      ^fxdiv@2020-04-17%gcc@13.1.0~ipo build_system=cmake build_type=Release generator=ninja arch=linux-ubuntu22.04-x86_64_v3
[+]      ^gloo@2023-05-19%gcc@13.1.0~cuda~ipo build_system=cmake build_type=Release generator=ninja arch=linux-ubuntu22.04-x86_64_v3
[+]      ^intel-oneapi-mkl@2023.2.0%gcc@13.1.0~cluster+envmods~ilp64+shared build_system=generic mpi_family=none threads=none arch=linux-ubuntu22.04-x86_64_v3
[+]          ^intel-tbb@2021.9.0%gcc@13.1.0~ipo+shared+tm build_system=cmake build_type=Release cxxstd=default generator=make patches=91755c6 arch=linux-ubuntu22.04-x86_64_v3
[^]      ^ninja@1.11.1%gcc@13.1.0+re2c build_system=generic arch=linux-ubuntu22.04-x86_64_v3
[^]          ^python@3.11.6%gcc@13.1.0+bz2+crypt+ctypes+dbm~debug+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tkinter+uuid+zlib build_system=generic patches=13fa8bf,b0615b2,ebdca64,f2fd060 arch=linux-ubuntu22.04-x86_64_v3
[^]              ^expat@2.5.0%gcc@13.1.0+libbsd build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[^]                  ^libbsd@0.11.7%gcc@13.1.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[^]                      ^libmd@1.0.4%gcc@13.1.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[^]              ^gettext@0.22.3%gcc@13.1.0+bzip2+curses+git~libunistring+libxml2+pic+shared+tar+xz build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[^]                  ^libxml2@2.10.3%gcc@13.1.0+pic~python+shared build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[^]              ^libffi@3.4.4%gcc@13.1.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[^]              ^libxcrypt@4.4.35%gcc@13.1.0~obsolete_api build_system=autotools patches=4885da3 arch=linux-ubuntu22.04-x86_64_v3
[^]              ^util-linux-uuid@2.38.1%gcc@13.1.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[^]          ^re2c@2.2%gcc@13.1.0 build_system=generic arch=linux-ubuntu22.04-x86_64_v3
[+]      ^numactl@2.0.14%gcc@13.1.0 build_system=autotools patches=4e1d78c,62fc8a8,ff37630 arch=linux-ubuntu22.04-x86_64_v3
[^]          ^autoconf@2.69%gcc@13.1.0 build_system=autotools patches=35c4492,7793209,a49dd5b arch=linux-ubuntu22.04-x86_64_v3
[^]          ^automake@1.16.5%gcc@13.1.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[^]          ^libtool@2.4.7%gcc@13.1.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[^]          ^m4@1.4.19%gcc@13.1.0+sigsegv build_system=autotools patches=9dc5fbd,bfdffa7 arch=linux-ubuntu22.04-x86_64_v3
[^]              ^diffutils@3.9%gcc@13.1.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[^]              ^libsigsegv@2.14%gcc@13.1.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]      ^openblas@0.3.24%gcc@13.1.0~bignuma~consistent_fpcsr+fortran~ilp64+locking+pic+shared build_system=makefile symbol_suffix=none threads=none arch=linux-ubuntu22.04-x86_64_v3
[^]          ^perl@5.38.0%gcc@13.1.0+cpanm+opcode+open+shared+threads build_system=generic patches=714e4d1 arch=linux-ubuntu22.04-x86_64_v3
[^]              ^berkeley-db@18.1.40%gcc@13.1.0+cxx~docs+stl build_system=autotools patches=26090f4,b231fcc arch=linux-ubuntu22.04-x86_64_v3
[+]      ^openmpi@4.1.6%gcc@13.1.0~atomics~cuda~cxx~cxx_exceptions~gpfs~internal-hwloc~internal-pmix~java~legacylaunchers~lustre~memchecker~openshmem~orterunprefix+romio+rsh~singularity+static+vt+wrapper-rpath build_system=autotools fabrics=none schedulers=none arch=linux-ubuntu22.04-x86_64_v3
[+]          ^hwloc@2.9.1%gcc@13.1.0~cairo~cuda~gl~libudev+libxml2~netloc~nvml~oneapi-level-zero~opencl+pci~rocm build_system=autotools libs=shared,static arch=linux-ubuntu22.04-x86_64_v3
[+]              ^libpciaccess@0.13.5%gcc@13.1.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[^]                  ^util-macros@1.19.3%gcc@13.1.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]              ^libxml2@2.9.10%gcc@13.1.0+pic~python+shared build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]          ^openssh@9.5p1%gcc@13.1.0+gssapi build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]              ^krb5@1.20.1%gcc@13.1.0+shared build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]                  ^bison@3.4.2%gcc@13.1.0 build_system=autotools patches=89aa362 arch=linux-ubuntu22.04-x86_64_v3
[^]                  ^findutils@4.9.0%gcc@13.1.0 build_system=autotools patches=440b954 arch=linux-ubuntu22.04-x86_64_v3
[^]              ^libedit@3.1-20210216%gcc@13.1.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[^]          ^pkgconf@1.9.5%gcc@13.1.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]          ^pmix@5.0.1%gcc@13.1.0~docs+pmi_backwards_compatibility~python~restful build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]              ^libevent@2.1.12%gcc@13.1.0+openssl build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[^]                  ^libtool@2.4.7%gcc@13.1.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[^]                      ^m4@1.4.19%gcc@13.1.0+sigsegv build_system=autotools patches=9dc5fbd,bfdffa7 arch=linux-ubuntu22.04-x86_64_v3
[^]                          ^libsigsegv@2.14%gcc@13.1.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]      ^protobuf@3.24.3%gcc@13.1.0~ipo+shared build_system=cmake build_type=Release generator=make arch=linux-ubuntu22.04-x86_64_v3
[+]          ^abseil-cpp@20230125.3%gcc@13.1.0~ipo+shared build_system=cmake build_type=Release cxxstd=14 generator=make arch=linux-ubuntu22.04-x86_64_v3
[+]      ^psimd@2020-05-17%gcc@13.1.0~ipo build_system=cmake build_type=Release generator=ninja arch=linux-ubuntu22.04-x86_64_v3
[+]      ^pthreadpool@2021-04-13%gcc@13.1.0~ipo build_system=cmake build_type=Release generator=ninja arch=linux-ubuntu22.04-x86_64_v3
[+]      ^py-astunparse@1.6.3%gcc@13.1.0 build_system=python_pip arch=linux-ubuntu22.04-x86_64_v3
[+]          ^py-six@1.16.0%gcc@13.1.0 build_system=python_pip arch=linux-ubuntu22.04-x86_64_v3
[+]      ^py-filelock@3.12.4%gcc@13.1.0 build_system=python_pip arch=linux-ubuntu22.04-x86_64_v3
[+]          ^py-hatch-vcs@0.3.0%gcc@13.1.0 build_system=python_pip arch=linux-ubuntu22.04-x86_64_v3
[+]          ^py-hatchling@1.18.0%gcc@13.1.0 build_system=python_pip arch=linux-ubuntu22.04-x86_64_v3
[+]              ^py-editables@0.3%gcc@13.1.0 build_system=python_pip arch=linux-ubuntu22.04-x86_64_v3
[+]              ^py-packaging@23.1%gcc@13.1.0 build_system=python_pip arch=linux-ubuntu22.04-x86_64_v3
[+]              ^py-pathspec@0.11.1%gcc@13.1.0 build_system=python_pip arch=linux-ubuntu22.04-x86_64_v3
[+]              ^py-pluggy@1.0.0%gcc@13.1.0 build_system=python_pip arch=linux-ubuntu22.04-x86_64_v3
[+]              ^py-trove-classifiers@2023.8.7%gcc@13.1.0 build_system=python_pip arch=linux-ubuntu22.04-x86_64_v3
[+]                  ^py-calver@2022.6.26%gcc@13.1.0 build_system=python_pip arch=linux-ubuntu22.04-x86_64_v3
[+]      ^py-fsspec@2023.1.0%gcc@13.1.0~http build_system=python_pip arch=linux-ubuntu22.04-x86_64_v3
[+]      ^py-jinja2@3.1.2%gcc@13.1.0~i18n build_system=python_pip arch=linux-ubuntu22.04-x86_64_v3
[+]          ^py-markupsafe@2.1.3%gcc@13.1.0 build_system=python_pip arch=linux-ubuntu22.04-x86_64_v3
[+]      ^py-networkx@2.7.1%gcc@13.1.0+default~extra build_system=python_pip arch=linux-ubuntu22.04-x86_64_v3
[+]      ^py-numpy@1.26.2%gcc@13.1.0 build_system=python_pip patches=873745d arch=linux-ubuntu22.04-x86_64_v3
[+]          ^py-cython@3.0.4%gcc@13.1.0 build_system=python_pip arch=linux-ubuntu22.04-x86_64_v3
[+]          ^py-pyproject-metadata@0.7.1%gcc@13.1.0 build_system=python_pip arch=linux-ubuntu22.04-x86_64_v3
[+]      ^py-pip@23.1.2%gcc@13.1.0 build_system=generic arch=linux-ubuntu22.04-x86_64_v3
[+]      ^py-protobuf@3.20.3%gcc@13.1.0~cpp build_system=python_pip arch=linux-ubuntu22.04-x86_64_v3
[+]      ^py-pybind11@2.11.0%gcc@13.1.0~ipo build_system=cmake build_type=Release generator=ninja arch=linux-ubuntu22.04-x86_64_v3
[+]      ^py-pyyaml@5.4.1%gcc@13.1.0+libyaml build_system=python_pip arch=linux-ubuntu22.04-x86_64_v3
[+]          ^libyaml@0.2.5%gcc@13.1.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]      ^py-requests@2.31.0%gcc@13.1.0~socks build_system=python_pip arch=linux-ubuntu22.04-x86_64_v3
[+]          ^py-certifi@2023.7.22%gcc@13.1.0 build_system=python_pip arch=linux-ubuntu22.04-x86_64_v3
[+]          ^py-charset-normalizer@3.3.0%gcc@13.1.0 build_system=python_pip arch=linux-ubuntu22.04-x86_64_v3
[+]          ^py-idna@3.4%gcc@13.1.0 build_system=python_pip arch=linux-ubuntu22.04-x86_64_v3
[+]          ^py-urllib3@2.0.6%gcc@13.1.0~brotli~secure~socks build_system=python_pip arch=linux-ubuntu22.04-x86_64_v3
[+]      ^py-setuptools@68.0.0%gcc@13.1.0 build_system=generic arch=linux-ubuntu22.04-x86_64_v3
[+]      ^py-sympy@1.11.1%gcc@13.1.0 build_system=python_pip arch=linux-ubuntu22.04-x86_64_v3
[+]          ^py-mpmath@1.2.1%gcc@13.1.0 build_system=python_pip arch=linux-ubuntu22.04-x86_64_v3
[+]      ^py-tqdm@4.66.1%gcc@13.1.0~notebook~telegram build_system=python_pip arch=linux-ubuntu22.04-x86_64_v3
[+]          ^py-setuptools-scm@7.1.0%gcc@13.1.0+toml build_system=python_pip arch=linux-ubuntu22.04-x86_64_v3
[+]              ^git@2.42.0%gcc@13.1.0+man+nls+perl+subtree~svn~tcltk build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[^]                  ^pcre2@10.42%gcc@13.1.0~jit+multibyte build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]              ^py-tomli@2.0.1%gcc@13.1.0 build_system=python_pip arch=linux-ubuntu22.04-x86_64_v3
[+]      ^py-typing-extensions@4.8.0%gcc@13.1.0 build_system=python_pip arch=linux-ubuntu22.04-x86_64_v3
[+]          ^py-flit-core@3.9.0%gcc@13.1.0 build_system=python_pip arch=linux-ubuntu22.04-x86_64_v3
[+]      ^py-wheel@0.41.2%gcc@13.1.0 build_system=generic arch=linux-ubuntu22.04-x86_64_v3
[+]      ^python@3.11.6%gcc@13.1.0+bz2+crypt+ctypes+dbm~debug+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tkinter+uuid+zlib build_system=generic patches=13fa8bf,b0615b2,ebdca64,f2fd060 arch=linux-ubuntu22.04-x86_64_v3
[^]          ^bzip2@1.0.8%gcc@13.1.0~debug~pic+shared build_system=generic arch=linux-ubuntu22.04-x86_64_v3
[^]          ^expat@2.5.0%gcc@13.1.0+libbsd build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[^]              ^libbsd@0.11.7%gcc@13.1.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[^]                  ^libmd@1.0.4%gcc@13.1.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[^]          ^gdbm@1.23%gcc@13.1.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]          ^gettext@0.22.3%gcc@13.1.0+bzip2+curses+git~libunistring+libxml2+pic+shared+tar+xz build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[^]              ^libiconv@1.17%gcc@13.1.0 build_system=autotools libs=shared,static arch=linux-ubuntu22.04-x86_64_v3
[^]              ^tar@1.34%gcc@13.1.0 build_system=autotools zip=pigz arch=linux-ubuntu22.04-x86_64_v3
[^]                  ^pigz@2.7%gcc@13.1.0 build_system=makefile arch=linux-ubuntu22.04-x86_64_v3
[^]                  ^zstd@1.5.5%gcc@13.1.0+programs build_system=makefile compression=none libs=shared,static arch=linux-ubuntu22.04-x86_64_v3
[^]          ^libffi@3.4.4%gcc@13.1.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[^]          ^libxcrypt@4.4.33%gcc@13.1.0~obsolete_api build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[^]              ^perl@5.36.0%gcc@13.1.0+cpanm+open+shared+threads build_system=generic arch=linux-ubuntu22.04-x86_64_v3
[^]                  ^berkeley-db@18.1.40%gcc@13.1.0+cxx~docs+stl build_system=autotools patches=26090f4,b231fcc arch=linux-ubuntu22.04-x86_64_v3
[^]                  ^bzip2@1.0.8%gcc@13.1.0~debug~pic+shared build_system=generic arch=linux-ubuntu22.04-x86_64_v3
[^]                      ^diffutils@3.9%gcc@13.1.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[^]                          ^libiconv@1.17%gcc@13.1.0 build_system=autotools libs=shared,static arch=linux-ubuntu22.04-x86_64_v3
[^]                  ^gdbm@1.23%gcc@13.1.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[^]                      ^readline@8.2%gcc@13.1.0 build_system=autotools patches=bbf97f1 arch=linux-ubuntu22.04-x86_64_v3
[^]                          ^ncurses@6.4%gcc@13.1.0~symlinks+termlib abi=none build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[^]                  ^zlib@1.2.13%gcc@13.1.0+optimize+pic+shared build_system=makefile arch=linux-ubuntu22.04-x86_64_v3
[^]          ^openssl@3.1.3%gcc@13.1.0~docs+shared build_system=generic certs=mozilla arch=linux-ubuntu22.04-x86_64_v3
[^]              ^ca-certificates-mozilla@2023-05-30%gcc@13.1.0 build_system=generic arch=linux-ubuntu22.04-x86_64_v3
[^]          ^readline@8.2%gcc@13.1.0 build_system=autotools patches=bbf97f1 arch=linux-ubuntu22.04-x86_64_v3
[^]          ^sqlite@3.43.2%gcc@13.1.0+column_metadata+dynamic_extensions+fts~functions+rtree build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[^]          ^util-linux-uuid@2.38.1%gcc@13.1.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[^]          ^xz@5.4.1%gcc@13.1.0~pic build_system=autotools libs=shared,static arch=linux-ubuntu22.04-x86_64_v3
[+]      ^sleef@3.5.1_2020-12-22%gcc@13.1.0~ipo build_system=cmake build_type=Release generator=ninja arch=linux-ubuntu22.04-x86_64_v3
[+]      ^valgrind@3.20.0%gcc@13.1.0+boost+mpi+only64bit~ubsan build_system=autotools libs=shared,static arch=linux-ubuntu22.04-x86_64_v3
[+]          ^boost@1.83.0%gcc@13.1.0+atomic+chrono~clanglibcpp~container~context~contract~coroutine~date_time~debug+exception~fiber~filesystem~graph~graph_parallel~icu~iostreams~json~locale~log~math~mpi+multithreaded~nowide~numpy~pic~program_options~python~random~regex~serialization+shared~signals~singlethreaded~stacktrace+system~taggedlayout~test+thread~timer~type_erasure~versionedlayout~wave build_system=generic cxxstd=98 patches=a440f96 visibility=hidden arch=linux-ubuntu22.04-x86_64_v3

```

### Error message

<details><summary>Error message</summary>

<pre>
-- /software/spack_environments/scratch/fg12/play1/lib/spack/env/gcc/g++ /tmp/fg12/spack-stage/spack-stage-py-torch-2.1.2-yvhzuit55rplsbknta7ssjijlam5bqvn/spack-src/torch/abi-check.cpp -o /tmp/fg12/spack-stage/spack-stage-py-torch-2.1.2-yvhzuit55rplsbknta7ssjijlam5bqvn/spack-src/build/abi-check
  -- Determined _GLIBCXX_USE_CXX11_ABI=1
  -- Could not find ccache. Consider installing ccache to speed up compilation.
  -- Performing Test C_HAS_AVX_1
  -- Performing Test C_HAS_AVX_1 - Success
  -- Performing Test C_HAS_AVX2_1
  -- Performing Test C_HAS_AVX2_1 - Success
  -- Performing Test C_HAS_AVX512_1
  -- Performing Test C_HAS_AVX512_1 - Failed
  -- Performing Test C_HAS_AVX512_2
  -- Performing Test C_HAS_AVX512_2 - Failed
  -- Performing Test C_HAS_AVX512_3
  -- Performing Test C_HAS_AVX512_3 - Failed
  -- Performing Test CXX_HAS_AVX_1
  -- Performing Test CXX_HAS_AVX_1 - Success
  -- Performing Test CXX_HAS_AVX2_1
  -- Performing Test CXX_HAS_AVX2_1 - Success
  -- Performing Test CXX_HAS_AVX512_1
  -- Performing Test CXX_HAS_AVX512_1 - Failed
  -- Performing Test CXX_HAS_AVX512_2
  -- Performing Test CXX_HAS_AVX512_2 - Failed
  -- Performing Test CXX_HAS_AVX512_3
  -- Performing Test CXX_HAS_AVX512_3 - Failed
</pre></details>

### Information on your system

**Spack.yaml**

```
# This is a Spack Environment file.
#
# It describes a set of packages to be installed, along with
# configuration settings.
spack:
  # add package specs to the `specs` list
  specs:
    - py-numpy@1.26.2 ^python@3.10.13
    - matrix:
      - [ 'py-torch@2.1.2' ]
      - [ '^python@3.10.13' ]
      - [ '~cuda' ]
  view: false
  concretizer:
    unify: false
```

**System Information**
```
spack debug report
* **Spack:** 0.21.0 (65d3221a9c436e77af9e802e543c71bd702ff4e2)
* **Python:** 3.10.12
* **Platform:** linux-ubuntu22.04-broadwell
* **Concretizer:** clingo
```
**Package Maintainers**
@adamjstewart 

### Additional information

[spack-build-env-mods.txt](https://github.com/spack/spack/files/14020865/spack-build-env-mods.txt)
[spack-build-env.txt](https://github.com/spack/spack/files/14020866/spack-build-env.txt)
[spack-build-out.txt](https://github.com/spack/spack/files/14020867/spack-build-out.txt)
[spack-build-01-install-out.txt](https://github.com/spack/spack/files/14020868/spack-build-01-install-out.txt)

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

**Comment by adamjstewart** [comment]
I believe this bug was fixed by #40893 and made obsolete by #41508, can you try again on the tip of develop?

**Comment by fabiosanger** [comment]
thanks for the suggestion. it builds fine
